src/client/resources/conversions.ts

```diff
@@ -682,6 +682,50 @@ export function createMigHook(migHook: any, namespace: string) {
   };
 }
 
+export function createMigTokenSecret(name: string, namespace: string, rawToken: string) {
+  // btoa => to base64, atob => from base64
+  const encodedToken = btoa(rawToken);
+  return {
+    apiVersion: 'v1',
+    data: {
+      token: encodedToken,
+    },
+    kind: 'Secret',
+    metadata: {
+      generateName: `${name}-`,
+      namespace,
+    },
+    type: 'Opaque',
+  };
+}
+
+export function createMigToken(
+  name: string,
+  namespace: string,
+  migTokenSecretName: string,
+  migTokenSecretNamespace: string,
+  migClusterName: string
+) {
+  return {
+    apiVersion: 'migration.openshift.io/v1alpha1',
+    kind: 'MigToken',
+    metadata: {
+      name,
```

Comment:
should we use the generateName field for the token name too?

Reply:
Don't think so. It's a primary Mig object like a MigCluster or MigStorage object and the user provides a name. We've discussed maybe decoupling the name the user provides from the actual name that's used on the object so it doesn't have to abide by kubernetes naming requirements as a primary key, but I think that's distinctly separate from this.